tests/unit_tests/FixHGVS.php

```diff
@@ -186,103 +206,56 @@ public static function dataProviderFixHGVS ()
             array('(c.(123_125)insA)', 'c.((123_125)insA)'),
 
             // Redundant parentheses.
-            array('c.(1_2)insA', 'c.1_2insA'),
             array('c.1_2ins(A)', 'c.1_2insA'),
-            array('c.1_2ins[A]', 'c.1_2insA'),
-
-            // Missing parentheses.
-            array('c.1_100insA', 'c.(1_100)insA'),
-            array('c.1_100del(10)', 'c.(1_100)del(10)'),
 
             // Wrongly placed suffixes.
             array('c.123delA', 'c.123del'),
-            array('c.(1_100)del(20)', 'c.(1_100)del(20)'),
+
+            // Wrongly formatted suffixes
+            array('c.1_2ins[A]', 'c.1_2insA'),
 
             // Redundant question marks.
             array('g.?del', 'g.?del'),
             array('g.1_?del', 'g.1_?del'),
-            array('g.?_100del', 'g.?_100del'),  // Fixme; have another look.
+            array('g.?_100del', 'g.?_100del'),
             array('g.?_?del', 'g.?del'),
             array('g.(?_?)del', 'g.?del'),
-            array('g.(?_5)_10del', 'g.(?_5)_10'),
-            array('g.(5_?)_10del', 'g.(5_?)_10del'), // Fixme; have another look.
+
+            array('g.(?_5)_10del', 'g.(?_5)_10del'),
+            array('g.(5_?)_10del', 'g.(5_?)_10del'),
+            array('g.(5_?)_?del', 'g.(5_?)del'),
             array('g.(?_?)_10del', 'g.?_10del'),
+
             array('g.5_(10_?)del', 'g.5_(10_?)del'),
-            array('g.5_(?_10)del', 'g.5_(?_10)del'), // Fixme; have another look.
+            array('g.5_(?_10)del', 'g.5_(?_10)del'),
+            array('g.?_(?_10)del', 'g.(?_10)del'),
             array('g.5_(?_?)del', 'g.5_?del'),
+
             array('g.(?_5)_(10_?)del', 'g.(?_5)_(10_?)del'),
-            array('g.(5_?)_(?_10)del', 'g.(5_10)del'),
-            array('g.(?_?)_(?_?)del', 'g.?del'),
+            array('g.(5_?)_(?_10)del', 'g.(5_?)_(?_10)del'),
+            array('g.(5_?)_(?_10)del(3)', 'g.(5_10)del(3)'),
```

Comment:
This seems to be in conflict with the example two lines above this one. Is `g.(5_?)_(?_10)` replaced into `g.(5_10)` or not? Does it depend on the suffix? Although I can see the logic behind that, I also remember Johan saying `g.(5_?)_(?_10)` was always incorrect and it should be replaced to `g.(5_10)`. If `g.(5_10)del` is not correct HGVS, we should check with him just to be sure.